Feat(eos_designs): Enable evpn_gateway for pathfinder deployment using next-hop-self (#5082)
aristanetworks/ansible-avd: ansible_collections/arista/avd/roles/eos_designs/docs/how-to/wan.md

- All the WAN routers must have a common path-group with at least one WAN route server to be able to inject the default control-plane match statement in the VRF default WAN policy.
- For the default VRF, routes received over BGP peering configured under tenants in `network_services` will not be automatically advertised to the WAN (they will be advertised toward the LAN if eBGP is used). To advertise them towards the WAN, they need to be injected in EVPN and this can be achieved by adding a route-map to mark them with the site SOO.
- Internet exit policies are not supported under WAN port-channel interfaces.

> - EVPN WAN gateway is supported only on sites with single WAN Router.

Reviewer: let's move this to the previous section (preview) as it requires to use a key in preview (`wan_use_evpn_node_settings_for_lan`) to be triggered.

A section about this should be added int the how-to as well around the LAN part

    It is *not* possible to use multiple direct HA links while disabling the port-channel.

#### EVPN Gateway LAN [PREVIEW]

- the LAN routes are received via EVPN
- Enabling the gateway requires to configure:

>   - `wan_use_evpn_node_settings_for_lan: True`

Reviewer: ```suggestion
  - `wan_use_evpn_node_settings_for_lan: true`
```

  - `overlay_routing_protocol: ebgp` for the WAN router
  - `evpn_role: client` for the WAN router
  - an EVPN route server should be defined, e.g. using `evpn_route_servers` settings.

> - When all the conditions are met, the WAN router configuration will add the Gateway parameter and inject the following configuration:

Reviewer: ```suggestion
- When all the conditions are met, the EVPN Gateway using DPS Interconnect is enabled and the following configuration is added to the WAN router:
```

    ```bash
    router adaptive-virtual-topology

>        topology role edge gateway vxlan # Notice the gateway VXLAN

Reviewer: ```suggestion
       topology role edge gateway vxlan # (1)!
```

       [...]
    [...]
    router bgp <AS>
       ...
       !
       address-family evpn
          neighbor WAN_RR activate
          neighbor WAN_RR domain remote
          neighbor WAN_RR encapsulation path-selection
          neighbor DC1_RR activate
          neighbor default next-hop-self received-evpn-routes route-type ip-prefix inter-domain
    ```

> 

Reviewer: ```suggestion
    ```

1. Notice the gateway VXLAN

```
